src/entity_worker.py

```diff
@@ -0,0 +1,650 @@
+                                         , user_info:dict, property_key:str=None) -> dict:
+
+        # Use the internal token to query the target entity to assure it is returned. This way public
+        # entities can be accessed even if valid_user_token is None.
+        internal_token = self.authHelper.getProcessSecret()
+        entity_dict = self._query_target_entity(entity_id=entity_id)
+        normalized_entity_type = entity_dict['entity_type']
+
+        # Get the generated complete entity result from cache if exists
+        # Otherwise re-generate on the fly
+        complete_dict = self.schemaMgr.get_complete_entity_result(token=internal_token
+                                                                  , entity_dict=entity_dict)
+
+        # Determine if the entity is publicly visible base on its data, only.
+        # To verify if a Collection is public, it is necessary to have its Datasets, which
+        # are populated as triggered data.  So pull back the complete entity for
+        # _get_entity_visibility() to check.
+        entity_scope = self._get_entity_visibility(entity_dict=complete_dict)
+        public_entity = (entity_scope is DataVisibilityEnum.PUBLIC)
+
+        # Initialize the user as authorized if the data is public.  Otherwise, the
+        # user is not authorized and credentials must be checked.
+        if public_entity:
+            user_authorized = True
+        else:
+            if valid_user_token is None:
+                raise entityEx.EntityForbiddenException(f"{normalized_entity_type} for {entity_id} is not"
+                                                        f" accessible without presenting a token.")
+
+            user_authorized = self._user_in_hubmap_read_group(user_info=user_info)
+            if not user_authorized:
+                raise entityEx.EntityForbiddenException(f"The requested {normalized_entity_type} has non-public data."
+                                                        f"  A Globus token with access permission is required.")
+
+        # We'll need to return all the properties including those generated by
+        # `on_read_trigger` to have a complete result e.g., the 'next_revision_uuid' and
+        # 'previous_revision_uuid' being used below.
+        # Collections, however, will filter out only public properties for return.
+
+        # Also normalize the result based on schema
+        final_result = self.schemaMgr.normalize_entity_result_for_response(complete_dict)
+
+        # Result filtering based on query string
+        # The `data_access_level` property is available in all entities Donor/Sample/Dataset
+        # and this filter is being used by gateway to check the data_access_level for file assets
+        # The `status` property is only available in Dataset and being used by search-api for revision
+        result_filtering_accepted_property_keys = ['data_access_level', 'status']
```

Comment:
The property filtering may not be generalized, I'd remove this feature from here. The current `/entities/<id>?property=data_access_level` is only being used for a very specific auth check in gateway and not even really that needed.

Reply:
Done. Property filtering support removed.